src/precice/bindings/matlab/+precice/@SolverInterface/SolverInterface.m

```diff
@@ -0,0 +1,283 @@
+classdef SolverInterface < handle
+    %SOLVERINTERFACE Matlab wrapper for the C++ SolverInterface
+    %   This class will serve as a gateway for the precice Solver Interface
+    %   in C++. It has access to a private mex Function written in the new
+    %   MATLAB C++ API. Such a function is actually an instance of a C++ 
+    %   class that is stored internally by MATLAB. Whenever they are 
+    %   "called", a subroutine of the object (operator ()) is invoked which
+    %   then has access to the members of the class.
+    %   Hence, we can store the actual solver interface in this class and
+    %   access it by invoking the mex function.
+    
+    properties(Access=private)
+        % Possible changes:
+        % - Allow the creation of multiple SolverInterfaces. For this, we
+        % should replace the interface pointer in the Gateway by a list of 
+        % interface pointers and add an InterfaceID as property of this
+        % class.
+    end
+    
+    methods
+        %% Construction and configuration
+        % Constructor
+        function obj = SolverInterface(SolverName,configFileName)
```

Comment:
We should extend the Interface with the arguments that are required for parallel runs: 

https://github.com/precice/precice/blob/b20780810baf4d36ed6bcd5715a27518643596e0/src/precice/SolverInterface.hpp#L70-L74

`solverProcessIndex`, `solverProcessSize` are missing. As far as I understand, the MATLAB bindings currently do not support parallel runs. We should still be consistent with the other language bindings in order to make the interface extensible.

In the current situation, we can simply throw an exception telling the user that parallel runs are not supported, if the user tries to start a parallel run. As soon as parallel runs are supported, we can roll out the feature and do not have to break the API.

Reply:
I added an issue on this: https://github.com/precice/matlab-bindings/issues/3